File: 1912_DP1/1912.cpp
#include <iostream>
#include <list>
using namespace std;

int main(){
    int num, max;
    cin >> num;
    int array[num], dp[num];
    for (int i=0; i<num; i++) {cin>>array[i];}
    max = dp[0] = array[0];
    for (int i=1; i<num; i++)
    {
        if (dp[i-1]+array[i]>array[i]) dp[i]=dp[i-1]+array[i];
        else dp[i] = array[i];
        if (dp[i] > max) max = dp[i];
    }
    cout << max;
    return 0;
}

File: 1912_DP1/1912.py
num = int(input())
arr = list(map(int,input().split()))
dp=[arr[0]]
for i in range(1,num): dp.append(dp[i-1]+arr[i]) if (dp[i-1]+arr[i]>arr[i]) else dp.append(arr[i])
print(max(dp))